### PR TITLE
- Added rendering FEN blocks (chess board notation)

### PR DESCRIPTION
**Pull request by cheeseonamonkey**
- Enabled rendering chess board w/ FEN notation *(using an open API, **fen2png**)*
- Created dir ` src/components/Chat/ChatContent/Message/CodeBlock/` and moved some components into their own files
	- new file:        src/components/Chat/ChatContent/Message/CodeBlock/CodeBar.tsx
	- moved file:   src/components/Chat/ChatContent/Message/CodeBlock/CodeBlock.tsx
	- new file         src/components/Chat/ChatContent/Message/CodeBlock/FENBlock.tsx
	- modified:     src/components/Chat/ChatContent/Message/View/ContentView.tsx
	
	
---

### Chess board [FEN](https://www.chess.com/terms/fen-chess) Renderer:

image

---

You must instruct in the prompt how to format the FEN code properly as a Markdown block _(format inspired by Obsidian.MD plugins like [dataview](https://github.com/blacksmithgu/obsidian-dataview#examples))_.

For example, this FEN code:
<pre>
```fen
rnbq1rk1/ppp1ppbp/3p1np1/8/2PPP3/2N2P2/PP2N1PP/R1BQKB1R b KQ - 0 10
```
</pre>

creates this chess board:
image

### CODE REVIEW

File: src/components/Chat/ChatContent/Message/CodeBlock/CodeBar.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import CopyIcon from "@icon/CopyIcon";
```
Comment (cheeseonamonkey) [inline src/components/Chat/ChatContent/Message/CodeBlock/CodeBar.tsx]: extracted this component from `CodeBlock.tsx`

File: src/components/Chat/ChatContent/Message/CodeBlock/FENBlock.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// This is a representation of a chess board, obtained by FEN notation.
```
Comment (cheeseonamonkey) [inline src/components/Chat/ChatContent/Message/CodeBlock/FENBlock.tsx]: chess FEN rendering component

**Comment by jackschedel** [comment]
lmfao this is so dumb I love it - GPT's are crazy bad at hallucinating when it comes to chess.

I'm down to PR this even though there is almost 0 chance that this will be useful with the current state of LLMs
(coming from a guy who's working on a chess engine as a side project lol)

**Comment by cheeseonamonkey** [comment]
>GPT's are crazy bad at hallucinating when it comes to chess

LOL good to know, I was not aware! :stuck_out_tongue_closed_eyes: 

Chess.com has an incredibly open API, and I wired it up to ChatGPT to generate report cards for games just because it's easy:
image

I found it to be vaguely accurate, but not particularly helpful :)

**Pull request closed without merging**